pkg/util/throttle.go

```diff
@@ -48,22 +53,36 @@ func NewFakeRateLimiter() RateLimiter {
 	return &fakeRateLimiter{}
 }
 
-func (t *tickRateLimiter) TryAccept() bool {
+func (t *tokenBucketRateLimiter) TryAccept() bool {
 	return t.limiter.TakeAvailable(1) == 1
 }
 
+func (t *tokenBucketRateLimiter) Saturation() float64 {
+	capacity := t.limiter.Capacity()
+	avail := t.limiter.Available()
+	// handles token burst, which means not saturated at all.
+	if avail > capacity {
+		avail = capacity
```

Comment:
hmm - to be honest, I think that what we should expose is not available/capacity, but available/(burst capacity)

WDYT?